hi-ml-histopathology/src/histopathology/utils/output_utils.py

```diff
@@ -324,8 +280,20 @@ def previous_validation_outputs_dir(self) -> Path:
     def test_outputs_dir(self) -> Path:
         return self.outputs_root / TEST_OUTPUTS_SUBDIR
 
+    def validate_slide_datasets_for_plot_options(
+        self, plot_options: Set[PlotOptionsKey], slides_dataset: Optional[SlidesDataset]
+    ) -> None:
+        if PlotOptionsKey.SLIDE_THUMBNAIL_HEATMAP in plot_options and not slides_dataset:
+            raise ValueError("You can not plot slide thumbnails and heatmaps without setting a slides_dataset. "
+                             "Please remove PlotOptionsKey.SLIDE_THUMBNAIL_HEATMAP from your plot options or provide "
+                             "a slide dataset.")
+
     def set_slides_dataset(self, slides_dataset: Optional[SlidesDataset]) -> None:
+        self.validate_slide_datasets_for_plot_options(self.test_plots_handler.plot_options, slides_dataset)
+        self.validate_slide_datasets_for_plot_options(self.val_plots_handler.plot_options, slides_dataset)
         self.slides_dataset = slides_dataset
```

Comment:
Does the `OutputsHandler` still need a reference to `slides_dataset`, or just the plots handlers?

Reply:
Good point, only the plots handlers!